liquigraph-core/src/main/java/org/liquigraph/core/configuration/validators/DatasourceConfigurationValidator.java

```diff
@@ -28,9 +29,20 @@
 
 public class DatasourceConfigurationValidator {
 
-    public Collection<String> validate(Optional<String> uri, Optional<DataSource> dataSource) {
-        if (uri.isPresent() == dataSource.isPresent()) {
-            return ImmutableList.of("Exactly one of JDBC URI or DataSource need to be configured");
+    public static final String INCORRECT_CONFIGURATION_OPTION_ERROR_MESSAGE = "Exactly one of JDBC URI, JDBC DataSource or GraphDatabaseService needs to be configured";
+
+    public Collection<String> validate(Optional<String> uri, Optional<DataSource> dataSource, Optional<GraphDatabaseService> database) {
+
+        int presentInstanceCount = 0;
+
+        for (Optional<?> optional : ImmutableList.of(uri, dataSource, database)) {
```

Comment:
I never used Guava, so you'll let me know if my new changes make sense. I couldn't find a simple way to filter based on the presence of an optional instance.